Public/Src/Tools/UnitTests/Analyzers/XLGToDBAnalyzerTests.cs

```diff
@@ -69,5 +74,64 @@ public void DBCreatedAndPopulated()
 
             XAssert.AreNotEqual(Directory.GetFiles(OutputDirPath.ToString(Context.PathTable), "*.sst").Length, 0);
         }
+
+        /// <summary>
+        /// This test makes sure that the number of events logged (per event type) match 
+        /// what is expected based on the build schedule that we manually populate and build.
+        /// </summary>
+        [Fact]
+        public void DBEventCountVerification()
+        {
+            Configuration.Logging.LogExecution = true;
+
+            var fileOne = FileArtifact.CreateOutputFile(Combine(TestDirPath, "foo.txt"));
+            var fileTwo = FileArtifact.CreateOutputFile(Combine(TestDirPath, "bar.txt"));
+            var dirOne = Path.Combine(ReadonlyRoot, "baz");
+            Directory.CreateDirectory(dirOne);
+            File.WriteAllText(Path.Combine(dirOne, "abc.txt"), "text");
+            File.WriteAllText(Path.Combine(dirOne, "xyz.txt"), "text12");
+
+            var pipA = CreateAndSchedulePipBuilder(new []
+            {
+                Operation.WriteFile(fileOne),
+            }).Process;
+
+            var pipB = CreateAndSchedulePipBuilder(new []
+            {
+                Operation.WriteFile(fileTwo),
+            }).Process;
+
+            var pipC = CreateAndSchedulePipBuilder(new []
+            {
+                Operation.ReadFile(fileOne),
+                Operation.WriteFile(CreateOutputFileArtifact())
+            }).Process;
+
+            var pipD = CreateAndSchedulePipBuilder(new []
+            {
+                Operation.EnumerateDir(new DirectoryArtifact(AbsolutePath.Create(Context.PathTable, dirOne), 0, false)),
+                Operation.WriteFile(CreateOutputFileArtifact())
+            }).Process;
+
+            var buildA = RunScheduler().AssertSuccess();
+
+            var analyzerRes = RunAnalyzer(buildA).AssertSuccess();
+
+            var dataStore = new XldbDataStore(storeDirectory: OutputDirPath.ToString(Context.PathTable));
+
+            XAssert.AreEqual(dataStore.GetFileArtifactContentDecidedEvents().Count(), LastGraph.AllFiles.Count());
+            XAssert.AreEqual(dataStore.GetWorkerListEvents().Count(), 0);
+            XAssert.AreEqual(dataStore.GetPipExecutionPerformanceEvents().Count(), LastGraph.RetrievePipReferencesOfType(PipType.Process).Count());

```

Comment:
can you extract `LastGraph.RetrievePipReferencesOfType(PipType.Process).Count());` into a local variable (e.g., `processPipCount` and use that variable here and elsewhere?  You can then use that value for `GetPipCacheMissEvents` as well.